learning/proofsearch.py

```diff
@@ -1305,9 +1305,9 @@ def test_proof_search(problemset='lean-library-logic',
 
 
 def make_agent(config, mle_log: MLELogger):
-    if config.get('agent_path'):
```

Comment:
this checks if agent_path is provided. dict.get without a default value defaults a default value of None. Instead we would get a key error every time calling make_agent if there is no agent_path provided.

Reply:
Are you sure that this work? https://github.com/omry/omegaconf/blob/6f9656bf091f6cb1ebfe22c926e60e1c9d25ca02/omegaconf/dictconfig.py#L428C5-L451

Reply:
Ah it returns None.